samples/zephyr/build.sh

```diff
@@ -22,14 +22,30 @@ fi
 
 GCC_ARM="$KONAN_DEPS/$TOOLCHAIN"
 
+rm -rf $DIR/build || exit 1
 mkdir -p $DIR/build && cd $DIR/build
 
-konanc $DIR/src/main.kt -target zephyr_$BOARD -linkerOpts -L$GCC_ARM/arm-none-eabi/lib/thumb -linkerOpts -lsupc++ -opt || exit 1
-
 export ZEPHYR_GCC_VARIANT=gccarmemb
 export GCCARMEMB_TOOLCHAIN_PATH=$GCC_ARM
 
 [ -f CMakeCache.txt ] || cmake -DCMAKE_VERBOSE_MAKEFILE=ON -DBOARD=$BOARD .. || exit 1
+
+# We need generated headers to be consumed by `cinterop`,
+# so we preconfigure the project with `make zephyr`.
+make zephyr
+
+. $DIR/c_interop/platforms/$BOARD.sh
+
+rm -f program.o
+
+mkdir -p $DIR/build/kotlin
+
+konanc $DIR/src/main.kt \
+        -target zephyr_$BOARD \
+        -r $DIR/c_interop/platforms/build \
+        -l $BOARD \
+        -opt -g -o $DIR/build/kotlin/program || exit 1
```

Comment:
-g?

Reply:
yep, to be useful as an interesting sample.